ganglia/verify_ganglia_running.py

```diff
@@ -0,0 +1,74 @@
+"""verify_ganglia_running.py: Script for ganglia initialization action test.
+"""
+import requests
+import socket
+import subprocess
+
+BASE = 'localhost'
+PORT = 80
+
+
+class GangliaApi(object):
+    def __init__(self, base, port):
+        self.path = 'http://{}:{}/ganglia/'.format(base, port)
+        self.host = socket.gethostname()
+        self.page = None
+        self.is_main_master = None
+
+    def validate_homepage(self):
+        if self.page:
+            if '-w-' in self.host:
+                raise Exception('NOK - Ganglia should not be running on worker node')
+            elif self.is_main_master:
+                if 'Cluster Report' in self.page:
+                    print('OK - Ganglia UI is running on master node')
+                else:
+                    raise Exception('NOK - Ganglia UI is not found on master node')
+        else:
+            if '-w-' in self.host:
+                print('OK - Ganglia is not running on worker node')
+            if '-m-' in self.host and not (self.is_main_master):
+                print('OK - Ganglia is not designed to be running on additional master nodes')
+
+    def get_homepage(self):
+        try:
+            self.page = requests.get(self.path).text[:200]
```

Comment:
A getter method should not mutate the state of an object -- this means that validate_homepage must come after get_homepage, which is confusing. Consider making GangliaApi purely an API client, and putting logic in separate functions.

Something like:

```
page = ganglia_api.get_page()
validate_homepage(page)
```

Same feedback with `detect_role()`.